mantidproject/mantid
# Detect add files without using -add extension

**Description of work.**
Detects add files without using the -add extension. This allows us to:

- Specify an added file without requiring the -add extension
- Load event nexus data that isn't added
- Refactor to some minor extent.

This enables us to next handle event data for transmission runs. For example SANS2D123-add and SAND2D-asdf will both be accepted

**To test:**
- Go to the unit test data direction (build/ExternalData/Testing/Data/UnitTest)
- Copy out `ZOOM9947-a` or set your data directory to the above
- Open the ISIS Sans Interface in Mantid
- Open the following user file
[USER_ZOOM[snip].txt](https://github.com/mantidproject/mantid/files/6305704/USER_ZOOM_Schoofs_4m_SampleChanger_193B_5x2mm_Small_BEAMSTOP_M5Trans_StandardOnly.txt)
- Enter `ZOOM9947-a` for the first 3 ~rows~ columns, hit process row
- The following error should appear and is expected (PR two resolves it): 
- `Cannot load event-mode data for transmission calculation`
- Try renaming ZOOM9947 to some other names including `ZOOM9947-add` they should all give the same above error


Part of #27482

*This does not require release notes because it doesn't fully resolve the problem. The follow-up PR will complete the work and include release notes*



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
